tests/offline-events-tests/RequestHandlerTests.swift

```diff
@@ -636,6 +636,78 @@ class RequestHandlerTests: XCTestCase {
+            if request.url?.absoluteString.contains(Const.Path.trackPushOpen) == true {
+                let sentAt = request.value(forHTTPHeaderField: "sentAt")
+                let createdAt = TestUtils.getRequestBody(request: request)?["createdAt"] as? Int
+                XCTAssertEqual(createdAt, Int(date.timeIntervalSince1970))
+                XCTAssertEqual(sentAt, "\(Int(date.timeIntervalSince1970))")
+                expectation1.fulfill()
+            }
+        }
+        let requestHandler = createRequestHandler(networkSession: networkSession,
+                                                  notificationCenter: MockNotificationCenter(),
+                                                  selectOffline: true)
+        requestHandler.trackPushOpen(NSNumber(value: campaignId),
+                                     templateId: NSNumber(value: templateId),
+                                     messageId: messageId,
+                                     appAlreadyRunning: appAlreadyRunning,
+                                     dataFields: dataFields,
+                                     onSuccess: nil,
+                                     onFailure: nil)
+        waitForTaskRunner(requestHandler: requestHandler, expectation: expectation1)
+    }
+
+    func testCreatedAtSentAtForOnline() throws {
+        let expectation1 = expectation(description: #function)
+        let date = Date().addingTimeInterval(-5000)
+        dateProvider.currentDate = date
+
+        let campaignId = 1
+        let templateId = 2
+        let messageId = "message_id"
+        let appAlreadyRunning = true
+        let dataFields: [String: Any] = [
+            "var1": "val1",
+            "var2": "val2",
+        ]
+
+        let networkSession = MockNetworkSession(statusCode: 200)
+        networkSession.requestCallback = { request in
+            let sentAt = request.value(forHTTPHeaderField: "sentAt")
+            XCTAssertEqual(sentAt, "\(Int(date.timeIntervalSince1970))")
+            let createdAt = TestUtils.getRequestBody(request: request)?["createdAt"] as? Int
+            XCTAssertEqual(createdAt, Int(date.timeIntervalSince1970))
+            expectation1.fulfill()
+        }
+        let requestHandler = createRequestHandler(networkSession: networkSession,
+                                                  notificationCenter: MockNotificationCenter(),
+                                                  selectOffline: false)
+        requestHandler.trackPushOpen(NSNumber(value: campaignId),
+                                     templateId: NSNumber(value: templateId),
+                                     messageId: messageId,
+                                     appAlreadyRunning: appAlreadyRunning,
+                                     dataFields: dataFields,
+                                     onSuccess: nil,
+                                     onFailure: nil)
+        wait(for: [expectation1], timeout: 15.0)
```

Comment:
Does this need to be `testExpectationTimeout`?

Reply:
Yes. I think that is more consistent.